src/projects/projects.service.ts

```diff
@@ -50,4 +108,17 @@ export class ProjectsService {
 
     return projects
   }
+
+  async update(
+    projectId: string,
+    { name, newClientId }: UpdateProjectDto,
+    currentClientId: string,
+    workspaceId: string,
+  ): Promise<Project> {
+    if (currentClientId === newClientId) {
+      return this.updateWithinSameClient(projectId, name, currentClientId)
+    } else {
```

Comment:
Combine everything in a single function. We don't need an extra function for the transfer